File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.cpp
class Solution {
public:
    int my(int i,vector<int>& cost)
    {
        if(i==0)return cost[0];
        if(i==1)return cost[1];
        int s=cost[i]+my(i-1,cost);
        int s2=cost[i]+my(i-2,cost);
        return min(s,s2);
    }
    int minCostClimbingStairs(vector<int>& cost) {
        int n=cost.size();
        return my(n,cost);
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-0.py
class Solution:
    def trap(self, height: List[int]) -> int:
        n=len(height)
        l = [0] * n
        r=[0]*n
        l[0]=height[0]
        r[n-1]=height[n-1]
        for i in range(1,n):
            l[i]=max(height[i],l[i-1])
        for i in range(n-2,-1,-1):
            r[i]=max(height[i],r[i+1])
        ans=0
        for i in range(n):
            ans+=min(l[i],r[i])-height[i]
        return ans
